manifests/functions/create_export.pp

```diff
@@ -49,7 +49,7 @@
       content => $line,
     }
 
-    if ! defined(File[$name]) {
+    unless File["${name}"] {
```

Comment:
This will fix your puppet-lint error, there is no need to quote a string that contains only a var
```suggestion
    unless File[$name] {
```